__tests__/Organization.js

```diff
@@ -1,150 +1,309 @@
 import '@testing-library/jest-dom/extend-expect';
+
+import { fireEvent, wait } from '@testing-library/react';
+import nock from 'nock';
+import { StatusCodes } from 'shared/constants';
 import {
   API_ENDPOINT,
-  AWSInfoResponse,
-  ORGANIZATION,
-  V4_CLUSTER,
   appCatalogsResponse,
   appsResponse,
+  AWSInfoResponse,
   generateRandomString,
+  getMockCall,
   getPersistedMockCall,
+  ORGANIZATION,
   orgResponse,
   orgsResponse,
   releasesResponse,
   userResponse,
+  V4_CLUSTER,
   v4AWSClusterResponse,
   v4AWSClusterStatusResponse,
   v4ClustersResponse,
 } from 'testUtils/mockHttpCalls';
-import { fireEvent, wait } from '@testing-library/react';
-import { StatusCodes } from 'shared/constants';
-import nock from 'nock';
 import { renderRouteWithStore } from 'testUtils/renderUtils';
 
 const BASE_ROUTE = '/organizations';
 
-const requests = {};
-
 // Responses to requests
-beforeAll(() => {
-  requests.userInfo = getPersistedMockCall('/v4/user/', userResponse);
-  requests.info = getPersistedMockCall('/v4/info/', AWSInfoResponse);
-  requests.organizations = getPersistedMockCall(
-    '/v4/organizations/',
-    orgsResponse
-  );
-  requests.organization = getPersistedMockCall(
-    `/v4/organizations/${ORGANIZATION}/`,
-    orgResponse
-  );
-  requests.clusters = getPersistedMockCall('/v4/clusters/', v4ClustersResponse);
-  requests.cluster = getPersistedMockCall(
-    `/v4/clusters/${V4_CLUSTER.id}/`,
-    v4AWSClusterResponse
-  );
-  requests.status = getPersistedMockCall(
+beforeEach(() => {
+  getMockCall('/v4/user/', userResponse);
+  getMockCall('/v4/info/', AWSInfoResponse);
+  getPersistedMockCall(`/v4/organizations/${ORGANIZATION}/`, orgResponse);
+  getMockCall('/v4/clusters/', v4ClustersResponse);
```

Comment:
Is there any reason why you only persist some calls?

Reply:
yes! I think this gives the tests more confidence that the requests are actually used only once.

Reply:
Don't follow. Then, why persist some of them? Why don't you just use always `getMockCall`?

Reply:
Yes good question! The ones where `getPersistedMockCall` is used, are called multiple times inside the tests, but since you've raised this, I'll change the tests to only use `getMockCall` :+1:

Reply:
Thanks. This is what I would expect since you are using `beforeEach`.

Reply:
There's `.times()` and some other sugary syntax for nock to specify how many times you expect the request to be made: https://github.com/nock/nock#repeat-response-n-times

Reply:
yea. I was being lazy.. :) I've tried the `.times()` function with our `get...MockCall` helpers but it didn't seem to be working.

Reply:
Yeah probably not possible since the .reply() is the last in the chain, and also we dont' return the nock thing anyways so chaining more onto it isn't possible even.

Reply:
Oh wait we do return, but yeah we'd still have to somehow add something into the middle of that chain...

```
export const getMockCall = (endpoint, response = []) =>
  nock(API_ENDPOINT)
    .get(endpoint)
    .reply(200, response);
```